pkg/secretless/config/v2/config.go

```diff
@@ -26,6 +27,38 @@ func (c Config) String() string {
 	return string(out)
 }
 
+// SecretlessYAML serializes Config to the secretless.yml format
+func (c Config) SecretlessYAML() ([]byte, error) {
```

Comment:
This really should be `Marshal` to align with other golang constructs (https://godoc.org/gopkg.in/yaml.v2#Marshal). I kind of wonder if we can even just implement https://godoc.org/gopkg.in/yaml.v2#Marshaler and use generic YAML marshalling.